File: Dot examples/ASP Book Examples/Book Examples/Chapter19/SqlHierarchicalDataSourceControlVB/App_Code/SqlHierarchicalEnumerable.vb
Imports System
Imports System.Web.UI
Imports System.Collections

Namespace AspNetUnleashed

    ''' <summary>
    ''' Represents a collection of SqlNodes
    ''' </summary>
    Public Class SqlHierarchicalEnumerable
        Inherits ArrayList
        Implements IHierarchicalEnumerable

        Public Sub New()
            MyBase.New()
        End Sub

        Public Function GetHierarchyData(ByVal EnumeratedItem As Object) As IHierarchyData _
        Implements IHierarchicalEnumerable.GetHierarchyData
            Return CType(EnumeratedItem, IHierarchyData)
        End Function

    End Class
End Namespace

File: Dot examples/ASP Book Examples/Book Examples/Chapter19/SqlHierarchicalDataSourceControlCS/App_Code/SqlHierarchicalEnumerable.cs
using System;
using System.Web.UI;
using System.Collections;

namespace AspNetUnleashed
{

    /// <summary>
    /// Represents a collection of SqlNodes
    /// </summary>
    public class SqlHierarchicalEnumerable : ArrayList, IHierarchicalEnumerable
    {

        public SqlHierarchicalEnumerable() : base() { }

        public IHierarchyData GetHierarchyData(object enumeratedItem)
        {
            return enumeratedItem as IHierarchyData;
        }



    }
}